kind/hack/tools/install_kuberay.sh

```diff
@@ -1,60 +1,67 @@
 #!/usr/bin/env bash
 
-op=$1
-
-source ../common.sh
+# This script manages the deployment of KubeRay components.
 
+# Define global variables
+op="$1"
 SLEEP_TIME="${SLEEP_TIME:-30}"
 MAX_RETRIES="${MAX_RETRIES:-5}"
-EXIT_CODE=0
 
+# Define functions
+
+# Function to deploy KubeRay
 deploy() {
-	sed -i.back "s/tag: v[0-9].*/tag: v${KUBERAY_APISERVER}/" ${ROOT_DIR}/hack/ray_api_server_values.yaml
-	helm repo add kuberay https://ray-project.github.io/kuberay-helm/
-	helm repo update
-	helm install kuberay-operator kuberay/kuberay-operator -n kuberay --version ${KUBERAY_OPERATOR} --set image.pullPolicy=IfNotPresent --create-namespace
-	helm install -f ${ROOT_DIR}/hack/ray_api_server_values.yaml kuberay-apiserver kuberay/kuberay-apiserver -n kuberay --version ${KUBERAY_APISERVER} --set image.pullPolicy=IfNotPresent
-	echo "Finished KubeRay deployment."
+    # Update YAML file with the desired version
+    sed -i.back "s/tag: v[0-9].*/tag: v${KUBERAY_APISERVER}/" "${ROOT_DIR}/hack/ray_api_server_values.yaml" || { echo "Error: Failed to update YAML file."; exit 1; }
+    
+    # Add Helm repo and update
+    helm repo add kuberay https://ray-project.github.io/kuberay-helm/ || { echo "Error: Failed to add Helm repo."; exit 1; }
+    helm repo update || { echo "Error: Failed to update Helm repo."; exit 1; }
+    
+    # Install kuberay-operator
+    helm install kuberay-operator kuberay/kuberay-operator -n kuberay --version "${KUBERAY_OPERATOR}" --set image.pullPolicy=IfNotPresent --create-namespace || { echo "Error: Failed to install kuberay-operator."; exit 1; }
+    
+    # Install kuberay-apiserver
+    helm install -f "${ROOT_DIR}/hack/ray_api_server_values.yaml" kuberay-apiserver kuberay/kuberay-apiserver -n kuberay --version "${KUBERAY_APISERVER}" --set image.pullPolicy=IfNotPresent || { echo "Error: Failed to install kuberay-apiserver."; exit 1; }
+    
+    echo "Finished KubeRay deployment."
 }
 
-wait(){
-	echo "Wait for kuberay deployment."
-	wait_for_pods "kuberay" "$MAX_RETRIES" "$SLEEP_TIME" || EXIT_CODE=$?
-
-	if [[ $EXIT_CODE -ne 0 ]]
-	then
-		echo "KubeRay Deployment unsuccessful. Not all pods running"
-		exit $EXIT_CODE
-	fi
+# Function to wait for KubeRay deployment
+wait_for_kuberay() {
+    echo "Waiting for KubeRay deployment..."
+    wait_for_pods "kuberay" "$MAX_RETRIES" "$SLEEP_TIME" || { echo "Error: KubeRay deployment unsuccessful. Not all pods running."; exit 1; }
```

Comment:
The command fails here:
./install_kuberay.sh: line 33: wait_for_pods: command not found,

you removed `source ../common.sh`, please return it